src/eva/language/models/wrappers/litellm.py

```
@@ -0,0 +1,77 @@
+    and includes an API key if provided.
+    """
+
+    def __init__(
+        self,
+        model_name_or_path: str,
+        model_kwargs: Dict[str, Any] | None = None,
+    ) -> None:
+        """Initializes the litellm chat model wrapper.
+
+        Args:
+            model_name_or_path: The model identifier (or name) for litellm
+                (e.g.,"openai/gpt-4o" or "anthropic/claude-3-sonnet-20240229").
+            model_kwargs: Additional keyword arguments to pass during
+                generation (e.g., `temperature`, `max_tokens`).
+        """
+        super().__init__()
+        self._model_name_or_path = model_name_or_path
+        self._model_kwargs = model_kwargs or {}
+        self.load_model()
+
+    @override
+    def load_model(self) -> None:
+        """Prepares the litellm model.
+
+        Note:
+            litellm doesn't require an explicit loading step; models are called
+            directly during generation. This method exists for API consistency.
+        """
+        pass
+
+    @override
+    def model_forward(self, prompts: List[str]) -> List[str]:
+        """Generates text using litellm.
+
+        Args:
+            prompts: A list of prompts to be converted into a "user" message.
+
+        Returns:
+            A list of generated text responses. Failed generations will contain
+            error messages instead of generated text.
+        """
+        messages = [[{"role": "user", "content": prompt}] for prompt in prompts]
+
+        responses = batch_completion(
+            model=self._model_name_or_path,
+            messages=messages,
+            **self._model_kwargs,
+        )
+
+        results = []
+        for i, response in enumerate(responses):
+            if isinstance(response, Exception):
+                error_msg = f"Error generating text for prompt {i}: {response}"
+                logger.error(error_msg)
+                results.append(f"[GENERATION ERROR]: {error_msg}")
```

I wonder if in this case we should raise the exception instead? Otherwise this could silently (not seeing the error prints in the log) fail and the evaluation compares ground truth with `[GENERATION ERROR] ...` strings?

---

The logic here was that I didn't want to abort the entire evaluation in case there was a failure in a single sample, which would be classified as "incorrect" anyway. But to be honest I also haven't seen it happening much, mostly in cases the server is not available or something along those lines. In this case it usually happens for all samples. So yes, let's do it!